backstage/plugins/knative-event-mesh-backend/src/providers/knativeEventMeshProcessor.test.ts

```diff
@@ -169,23 +182,108 @@ describe('KnativeEventMeshProcessor', () => {
                             'metadata.namespace': 'default',
                             'metadata.annotations.backstage.io/kubernetes-id': 'consumer-1',
                         },
+                        cursor: undefined,
+                        limit: 5
+                    },
+                    queryEntitiesResult: {
+                        items: [],
+                        pageInfo: {}
                     },
-                    queryEntitiesResult: [],
                 },
                 expectedRelations: [],
             },
+            {
+                name: 'should make a 2nd call with cursor if there are more than 5 items returned and limit is 5',
+                entity: {
+                    apiVersion: 'backstage.io/v1alpha1',
+                    kind: 'API',
+                    metadata: {
+                        namespace: 'default',
+                        name: 'et-1',
+                        consumedBy: ['consumer-1'],
+                    },
+                    spec: {
+                        owner: 'owner',
+                        system: 'system',
+                        lifecycle: 'lifecycle',
+                        definition: 'definition',
+                        type: 'eventType',
+                    },
+                },
+                query: {
+                    queryEntitiesRequest: {
+                        filter: {
+                            kind: 'component',
+                            'metadata.namespace': 'default',
+                            'metadata.annotations.backstage.io/kubernetes-id': 'consumer-1',
+                        },
+                        cursor: undefined,
+                        limit: 5,
+                    },
+                    queryEntitiesResult:
+                        {
+                          items: [
+                                {
```

Comment:
Maybe it makes more sense to be able to pass multiple queries that we want to mock. Something like:

```
    queries: [
       {
            queryEntitiesRequest: {
                filter: {...},
                cursor: undefined,
                limit: 5,
            },
            queryEntitiesResult: { 
                items: [...5 items],
                pageInfo: {
                    nextCursor: "AAAA",
                    prevCursor: "0"
               }
           }
        },
        {
            queryEntitiesRequest: {
                filter: {...},
                cursor: "AAAA",
                limit: 5,
            },
            queryEntitiesResult: { 
                items: [...2 items],
                pageInfo: {
                    nextCursor: undefined,
                    prevCursor: "AAAA"
               }
           }
        }
```

So, basically, in the suggestion above, I say that I want to mock 2 requests:
1. First receives 5 items and the cursor "AAAA"
2. Second one asks with the cursor "AAAA" and it receives 2 more items, with no next page cursor

What do you think?